lib/block-supports/layout.php

```diff
@@ -582,6 +582,24 @@ function gutenberg_incremental_id_per_prefix( $prefix = '' ) {
 	return $prefix . (string) ++$id_counters[ $prefix ];
 }
 
+/**
+ * Generates a unique ID based on the structure and values of a given array.
+ *
+ * This function serializes the array into a JSON string and generates a hash
+ * that serves as a unique identifier. Optionally, a prefix can be added to
+ * the generated ID for context or categorization.
+ *
+ * @param array  $data   The input array to generate an ID from.
+ * @param string $prefix Optional. A prefix to prepend to the generated ID. Default ''.
+ *
+ * @return string The generated unique ID for the array.
+ */
+function gutenberg_unique_prefixed_id_from_array( array $data, string $prefix = '' ): string {
```

Comment:
Nit (non-blocking): I think we could use a more generic name here, `wp_unique_id_from_values`. Since the prefix is optional, we could also remove it.

Reply:
I see the name was changed to `gutenberg_unique_id_from_values`. @DAreRodz, can you clarify your intentions regarding the prefix?